File: 07212023/47_165_Single_Letter_Count/java/SingleLetterCount.java
public class SingleLetterCount {
    public static void main(String[] args) {
        System.out.println(singleLetterCount("Hello World", 'h'));
        System.out.println(singleLetterCount("Hello World", 'z'));
        System.out.println(singleLetterCount("Hello World", 'l'));
    }

    public static int singleLetterCount(String inputString, char letter) {
        String lowerStringInput = inputString.toLowerCase(); 
        int counter = 0;   

        for (int i = 0; i < inputString.length(); i++) {
            if (lowerStringInput.charAt(i) == letter) {
                counter += 1;
            }
        }

        return counter;
    }
}

File: 07212023/47_165_Single_Letter_Count/python/single_letter_count.py
def single_letter_count(input_string: str, letter: str) -> int:
    """Takes in a word and a letter and return the count that letter appears in the word."""
    return input_string.lower().count(letter.lower())


print(single_letter_count.__doc__)
print(single_letter_count("Hello World", "h")) # 1
print(single_letter_count("Hello World", "z")) # 0
print(single_letter_count("HelLo World", "l")) # 3
